### PR TITLE
Add snow cover to map

### PR DESCRIPTION
**Pull request by developer-8**
The goal of this PR is to add "snow cover" to the map. To create a winter mood and bring variety to the game. Snow is simulated by lab-white tiles and therefore has some limitations compared to ordinary tiles, for example, snow does not have a transition with ordinary tiles, so special concrete tiles are used as transit tiles.

Features:
- Admins have the ability to enable and disable the snow cover through the config. Changes are applied only to the map of the next game.
- Several snow coverage modes, with different amounts of snow on the map + random mode
- Made better looking trees in areas with snow

image

Of course, snow cover will affect the performance of map generation, but this is a price that must be paid only once when generating a chunk. I did not do tests performance, but it was playable locally.

### CODE REVIEW

File: maps/biter_battles_v2/snow_cover.lua [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+			table_insert(tiles, {name = "lab-white", position = pos})
+			replace_trees[pos.x .. "_" .. pos.y] = true
+
+			::continue::
+		end
+	end
+
+	surface.set_tiles(tiles, true)
+
+	for _, entity in pairs(surface.find_entities({{left_top_x, left_top_y}, {left_top_x + 32, left_top_y + 32}})) do
```
Comment (TheBigZet) [inline maps/biter_battles_v2/snow_cover.lua:128]: Some weird indentation is happening here

File: maps/biter_battles_v2/snow_cover.lua [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+			if entity.name == "tree-02"
+				or entity.name == "tree-02-red"
+				or entity.name == "tree-03"
+				or entity.name == "tree-05"
+				or entity.name == "tree-07"
+			then
+				name = "tree-01"
+            end
+
+            if entity.name == "tree-08"
```
Comment (TheBigZet) [inline maps/biter_battles_v2/snow_cover.lua:141]: Maybe put the names in a list?

File: maps/biter_battles_v2/snow_cover.lua (hunk 1 above)
Comment (developer-8) [reply maps/biter_battles_v2/snow_cover.lua]: Fixed. When spaces and tabs are mixed, github doesn't show it the way my editor does

File: maps/biter_battles_v2/snow_cover.lua (hunk 2 above)
Comment (developer-8) [reply maps/biter_battles_v2/snow_cover.lua]: Сhanged using list

**Comment by developer-8** [comment]
I had to make `is_horizontal_border_river` global again, since now it is also called in another file

**Comment by Ragnarok77-factorio** [comment]
Tested during captain game, lag was noticed but it was very likely due to massive map reveal that killed ups and not the snow change. 
Neuro found that : 
image

But after discussion, it's just that there is no transition from snow to brick, so not a bug and nothing we can do about it.
I like the toggle option of the PR too.

**Review by TheBigZet — CHANGES_REQUESTED** [review]
I don't think it should be merged. Some players reported it being quite uncomfortable for eyes

**Comment by developer-8** [comment]
I am closing this PR as it is no longer relevant.

**Pull request closed without merging**